src/Elastic.Apm/Helpers/SystemInfoHelper.cs

```diff
@@ -161,27 +145,39 @@ protected virtual StreamReader GetCGroupAsStream()
 		internal const string PodUid = "KUBERNETES_POD_UID";
 		internal const string NodeName = "KUBERNETES_NODE_NAME";
 
-		internal KubernetesMetadata ParseKubernetesInfo(Container containerInfo, string hostName)
+		internal void ParseKubernetesInfo(Api.System system)
 		{
-			var @namespace = Environment.GetEnvironmentVariable(Namespace);
-			var podName = Environment.GetEnvironmentVariable(PodName);
-			var podUid = Environment.GetEnvironmentVariable(PodUid);
-			var nodeName = Environment.GetEnvironmentVariable(NodeName);
-
-			if (@namespace == null && podName == null && podUid == null && nodeName == null)
+			try
 			{
-				// By default, Kubernetes will set the hostname of the pod containers to the pod name.
-				// Users that override the name should use the Downward API to override the pod name.
-				return containerInfo != null
-					? new KubernetesMetadata { Pod = new Pod { Uid = containerInfo.Id, Name = hostName } }
-					: null;
-			}
+				var @namespace = Environment.GetEnvironmentVariable(Namespace);
+				var podName = Environment.GetEnvironmentVariable(PodName);
+				var podUid = Environment.GetEnvironmentVariable(PodUid);
+				var nodeName = Environment.GetEnvironmentVariable(NodeName);
+
+				if (@namespace != null || podName != null || podUid != null || nodeName != null)
+				{
+					system.Kubernetes ??= new KubernetesMetadata();
+					system.Kubernetes.Namespace = @namespace;
+
+					if (nodeName != null)
+						system.Kubernetes.Node = new Api.Kubernetes.Node { Name = nodeName };
 
-			var kubernetesMetadata = new KubernetesMetadata { Namespace = @namespace };
-			if (podName != null || podUid != null) kubernetesMetadata.Pod = new Pod { Name = podName, Uid = podUid };
-			if (nodeName != null) kubernetesMetadata.Node = new Api.Kubernetes.Node { Name = nodeName };
+					if (podUid != null || podName != null)
+					{
+						// retain existing pod values when environment variables are null
```

Comment:
Is this the intended line for this comment? 🤔 Seems a bit off to me. Here I'd either say 
>  retain existing pod values when environment variables are null, otherwise use them 

(so it indicates that the next lines will use them)
Or just move it to the end of the if block.

Reply:
The comment is in reference to the use of `??=`, to indicate that `new Pod()` is assigned _only_ when `system.Kubernetes.Pod` is null i.e. if it's not null, we want to retain any values that are already set, and will overwrite any existing values with `podUid` or `podName` from environment variables if they're not null.